core/modules/parsers/wikiparser/rules/codeblock.js

```diff
@@ -12,6 +12,25 @@ Wiki text rule for code blocks. For example:
 ```
 
 \*/
+
+/**
+ * Represents the `codeblock` rule.
+ * 
+ * @typedef {Object} CodeblockNode
+ * @property {string} type - The type of the widget, which is "codeblock".
+ * @property {Object} attributes - The attributes of the codeblock.
+ * @property {Object} attributes.code - The code attribute object.
+ * @property {string} attributes.code.type - The type of the code attribute, which is "string".
+ * @property {string} attributes.code.value - The actual code content within the code block.
+ * @property {number} attributes.code.start - The start position of the code in the source text.
+ * @property {number} attributes.code.end - The end position of the code in the source text.
+ * @property {Object} attributes.language - The language attribute object.
+ * @property {string} attributes.language.type - The type of the language attribute, which is "string".
+ * @property {string} attributes.language.value - The language specified after the triple backticks, if any.
+ * @property {number} attributes.language.start - The start position of the language string in the source text.
+ * @property {number} attributes.language.end - The end position of the language string in the source text.
+ */
+
```

Comment:
IMO having `@property {string} type ` and then `- The type of the widget, which is "codeblock". ` is the same thing. Both elements describe: "What it is". -- I think the second part should describe: "What it does", otherwise we can skip it. 

- @property does describe -- What it is
- Text describes -- What it does

eg: 

```

/**
 * Represents the parser `codeblock` rule.
 * 
 * @typedef {Object} CodeblockNode
 * @property {string} rule - Always "codeblock"
 * @property {string} type - Always "codeblock"
 * @property {number} start - Rule start marker in source text
 * @property {number} end - Rule end marker in source text
 * @property {Object} attributes - Contains "code" and "language" objects
 * @property {Object} attributes.code - The code attribute object
 * @property {string} attributes.code.type - Always "string"
 * @property {string} attributes.code.value - Actual code
 * @property {number} attributes.code.start - Start position of code in source text
 * @property {number} attributes.code.end - End position of code in source text
 * @property {Object} attributes.language - The language attribute object
 * @property {string} attributes.language.type - Always "string"
 * @property {string} attributes.language.value - Language specified after the triple backticks, if any
 * @property {number} attributes.language.start - Start position of the string in source text
 * @property {number} attributes.language.end - End position of the string in source text
 */
```

Should be as short as possible and still valid. Especially see `type, rule, start, end` which are properties of "CodeblockNode"

I did remove the "full stops" from all `@` elements. They are not needed
The first intro sentence has a full stop.

Reply:
Thanks, I'd copy that.

I have to admit, all jsdoc are generated by github copilot. I don't have time for this detail, I will leave it for future refinement. I'm still debugging the type when I'm using it in my plugin project.

Reply:
> I have to admit, all jsdoc are generated by github copilot. I don't have time for this detail, I will leave it for future refinement. I'm still debugging the type when I'm using it in my plugin project.

The new type information will significantly increase the TW code-size. So if there is redundant information it should be removed. 

And even more important it has to be valid. So if the co-pilot only adds comments in a more verbose form than the parameters are. There should not be any comments at all -- They have no value.

So if there are no comments, we actually know, that we have to add them manually. IMO for the tooltips it would be OK to start with the type info.

Reply:
> it has to be valid

This won't have problem, because I use the method body as input too. And we can auto merge "comment" type of PR based on https://github.com/TiddlyWiki/TiddlyWiki5/pull/7542 , so people can update comment quickly. I find this is quite frequent when maintaining https://github.com/tiddly-gittly/TW5-Typed